pkg/steps/pod.go

```diff
@@ -198,3 +170,79 @@ func PodStep(name string, config PodStepConfiguration, resources api.ResourceCon
 		jobSpec:     jobSpec,
 	}
 }
+
+func (s *podStep) getPodObject() (*coreapi.Pod, error) {
+	image := fmt.Sprintf("%s:%s", s.config.From.Name, s.config.From.Tag)
+	containerResources, err := resourcesFor(s.resources.RequirementsForStep(s.config.As))
+	if err != nil {
+		return nil, fmt.Errorf("unable to calculate %s pod resources for %s: %s", s.name, s.config.As, err)
+	}
+
+	if len(s.config.From.Namespace) > 0 {
+		return nil, fmt.Errorf("pod step does not supported an image stream tag reference outside the namespace")
+	}
+
+	pod := &coreapi.Pod{
+		ObjectMeta: meta.ObjectMeta{
+			Name: s.config.As,
+			Labels: trimLabels(map[string]string{
+				PersistsLabel:    "false",
+				JobLabel:         s.jobSpec.Job,
+				BuildIdLabel:     s.jobSpec.BuildId,
+				ProwJobIdLabel:   s.jobSpec.ProwJobID,
+				CreatedByCILabel: "true",
+			}),
+			Annotations: map[string]string{
+				JobSpecAnnotation:                     s.jobSpec.RawSpec(),
+				annotationContainersForSubTestResults: s.name,
+			},
+		},
+		Spec: coreapi.PodSpec{
+			ServiceAccountName: s.config.ServiceAccountName,
+			RestartPolicy:      coreapi.RestartPolicyNever,
+			Containers: []coreapi.Container{
+				{
+					Name:                     s.name,
+					Image:                    image,
+					Command:                  []string{"/bin/sh", "-c", "#!/bin/sh\nset -eu\n" + s.config.Commands},
+					Resources:                containerResources,
+					TerminationMessagePolicy: coreapi.TerminationMessageFallbackToLogsOnError,
+				},
+			},
+		},
+	}
+
+	if s.config.SecretName != "" {
+		pod.Spec.Containers[0].VolumeMounts = secretVolumeMountFromConfig(s.config)
+		pod.Spec.Volumes = secretVolumeFromConfig(s.config)
+	}
+	return pod, nil
+}
+
+func secretVolumeFromConfig(config PodStepConfiguration) []coreapi.Volume {
+	return []coreapi.Volume{
+		{
+			Name: testSecretName,
+			VolumeSource: coreapi.VolumeSource{
+				Secret: &coreapi.SecretVolumeSource{
+					SecretName: config.SecretName,
```

Comment:
pass only the `secretName` instead of the whole `PodStepConfiguration`

Reply:
I prefer this to be more unified so both functions take the same input... But change as per suggestion.